testing/scenario_app/android/app/src/androidTest/java/dev/flutter/scenariosui/ScreenshotUtil.java

```diff
@@ -0,0 +1,257 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package dev.flutter.scenariosui;
+
+import android.app.Activity;
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Xml;
+import androidx.test.InstrumentationRegistry;
+import androidx.test.runner.AndroidJUnitRunner;
+import androidx.test.runner.screenshot.Screenshot;
+import com.facebook.testing.screenshot.ScreenshotRunner;
+import com.facebook.testing.screenshot.internal.AlbumImpl;
+import com.facebook.testing.screenshot.internal.Registry;
+import com.facebook.testing.screenshot.internal.TestNameDetector;
+import dev.flutter.scenarios.TestableFlutterActivity;
+import java.io.BufferedOutputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import org.xmlpull.v1.XmlSerializer;
+
+/**
+ * Adapter for {@code com.facebook.testing.screenshot.Screenshot} that supports Flutter apps.
+ *
+ * <p>{@code com.facebook.testing.screenshot.Screenshot} relies on {@code View#draw(canvas)}, which
+ * doesn't draw Flutter's Surface or SurfaceTexture.
+ *
+ * <p>The workaround takes a full screenshot of the device and removes the status and action bars.
+ */
+public class ScreenshotUtil {
+  private XmlSerializer mXmlSerializer;
+  private AlbumImpl album;
+  private OutputStream mOutputStream;
+
+  private static ScreenshotUtil instance;
+  private static int BUFFER_SIZE = 1 << 16; // 64K
+
+  private static ScreenshotUtil getInstance() {
+    synchronized (ScreenshotUtil.class) {
+      if (instance == null) {
+        instance = new ScreenshotUtil();
+      }
+      return instance;
+    }
+  }
+
+  /** Starts the album, which contains the screenshots in a zip file, and a metadata.xml file. */
+  void init() {
+    if (mXmlSerializer != null) {
+      return;
+    }
+    album = AlbumImpl.create(Registry.getRegistry().instrumentation.getContext(), "default");
+    album.cleanup();
+
+    mXmlSerializer = Xml.newSerializer();
+    try {
+      mOutputStream =
+          new BufferedOutputStream(new FileOutputStream(album.getMetadataFile()), BUFFER_SIZE);
+    } catch (FileNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+    try {
+      mXmlSerializer.setOutput(mOutputStream, "utf-8");
+      mXmlSerializer.startDocument("utf-8", null);
+      // Start tag <screenshots>.
+      mXmlSerializer.startTag(null, "screenshots");
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  void writeText(String tagName, String value) throws IOException {
+    if (mXmlSerializer == null) {
+      throw new RuntimeException("ScreenshotUtil must be initialized.");
```

Comment:
We can add something like "..., call `onCreate...` to initialize" to direct the caller to the right path.